junebug/channel.py

```diff
@@ -0,0 +1,149 @@
+import collections
+from copy import deepcopy
+import json
+import uuid
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.web import http
+from vumi.service import WorkerCreator
+from vumi.servicemaker import VumiOptions
+
+from junebug.error import JunebugError
+
+
+class ChannelNotFound(JunebugError):
+    '''Raised when a channel's data cannot be found.'''
+    name = 'ChannelNotFound'
+    description = 'channel not found'
+    code = http.NOT_FOUND
+
+
+class InvalidChannelType(JunebugError):
+    '''Raised when an invalid channel type is specified'''
+    name = 'InvalidChannelType',
+    description = 'invalid channel type'
+    code = http.BAD_REQUEST
+
+
+transports = {
+    'telnet': 'vumi.transports.telnet.TelnetServerTransport',
+    'xmpp': 'vumi.transports.xmpp.XMPPTransport',
+}
```

Comment:
Update from RL discussion: configurable as in the ability to specify transport name to python path mapping in config file, using the dict above as the default. Not super important, so we are leaving it as is for now.